File: DTAtoSEDAS/DatToSedasReengineer/ConvertDatToSedas.vb
Imports System.IO
Imports Microsoft.VisualBasic.CompilerServices

Public Class ConvertDATtoSEDAS
    Private _SourcePath As String

    Private _SourceData As String

    Private _DestinationPath As String

    Private _DestinationData As String

    Private _SedasHeader As String

    Private _SedasFooter As String

    Private _BlockHeader As String

    Private _DATContent As String(,)

    Private _DataSets As Integer

    Private _Customers As Integer

    Private _SummeGes As Integer

    Private _ListDelCustomer As List(Of String)

    Private _ListDelArticle As List(Of String)

    Private _Counter As Integer

    Private _ErstelldatumSedas As String

    Public Sub New(src As String, dst As String, cnt As Integer, Optional dC As List(Of String) = Nothing, Optional dA As List(Of String) = Nothing)
        Me._SourcePath = ""
        Me._SourceData = ""
        Me._DestinationPath = ""
        Me._DestinationData = ""
        Me._SedasHeader = ""
        Me._SedasFooter = ""
        Me._BlockHeader = ""
        Me._DataSets = 0
        Me._Customers = 0
        Me._SummeGes = 0
        Me._Counter = 0
        Me._ErstelldatumSedas = ""
        Me._SourcePath = src
        Me._DestinationPath = dst
        Me._Counter = cnt
        Me._ListDelArticle = dA
        Me._ListDelCustomer = dC
    End Sub

    Public Function ConvertFile() As Boolean
        Dim result As Boolean = False
        LogMessage.LogOnly("Beginn der Konvertierung...")
        Me._ErstelldatumSedas = Me.ReverseDate(Strings.Mid(DateAndTime.Now.ToString(), 1, 10).Replace(".", "").Remove(4, 2))
        Try
            Dim flag As Boolean = File.Exists(Me._SourcePath)
            If Not flag Then
                result = False
                Return result
            End If
            LogMessage.LogOnly("Einlesen der Quelldatei...")
            Using streamReader As StreamReader = New StreamReader(Me._SourcePath)
                Me._SourceData = streamReader.ReadToEnd()
            End Using
        Catch expr_9B As Exception
            ProjectData.SetProjectError(expr_9B)
            result = False
            ProjectData.ClearProjectError()
            Return result
        End Try
        Dim array As String() = Strings.Split(Me._SourceData, vbCrLf, -1, CompareMethod.Binary)
        Dim num As Integer = 0
        Dim upperBound As Integer = array.GetUpperBound(0)
        ' The following expression was wrapped in a checked-statement
        For i As Integer = 0 To upperBound
            Dim flag2 As Boolean = Operators.CompareString(array(i), "", False) <> 0
            If flag2 Then
                num += 1
            End If
        Next
        Dim array2 As String() = New String(num + 1 - 1) {}
        Dim num2 As Integer = 0
        Dim upperBound2 As Integer = array.GetUpperBound(0)
        For j As Integer = 0 To upperBound2
            Dim flag3 As Boolean = Operators.CompareString(array(j), "", False) <> 0
            If flag3 Then
                array2(num2) = array(j)
                num2 += 1
            End If
        Next
        array = array2
        Try
            Dim flag4 As Boolean = Operators.CompareString(Strings.Mid(array(0), 1, 2), "NF", False) = 0
            If flag4 Then
                LogMessage.LogOnly("Einlesen neues Dateiformat...")
                Me._DATContent = Me.ReadNewDATData(array, Me._ErstelldatumSedas)
            Else
                LogMessage.LogOnly("Einlesen altes Dateiformat...")
                Me._DATContent = Me.ReadDATData(array, Me._ErstelldatumSedas)
            End If
            Dim flag5 As Boolean = Not Information.IsNothing(Me._DATContent) And Me.WriteSedasData()
            If flag5 Then
                LogMessage.LogOnly("Konvertierung in SEDAS.DAT abgeschlossen.")
                result = True
            End If
        Catch expr_1D8 As Exception
            ProjectData.SetProjectError(expr_1D8)
            Dim ex As Exception = expr_1D8
            LogMessage.LogOnly("Fehler beim Konvertieren in Sedas.dat." & vbCrLf + ex.ToString())
            result = False
            ProjectData.ClearProjectError()
        End Try
        Return result
    End Function

    Private Function ReadDATData(arrSourceLines As String(), ErstelldatumTTMMJJ As String) As String(,)
        Dim flag As Boolean = Operators.CompareString(arrSourceLines(0), "", False) = 0
        ' The following expression was wrapped in a checked-statement
        Dim array As String(,)
        If flag Then
            array = Nothing
            array = array
        Else
            Try
                Dim num As Integer = -1
                Dim upperBound As Integer = arrSourceLines.GetUpperBound(0)
                For i As Integer = 0 To upperBound
                    Dim flag2 As Boolean = Operators.CompareString(arrSourceLines(i), "", False) <> 0
                    If flag2 Then
                        num += 1
                    End If
                Next
                Dim array2 As String(,) = New String(num + 1 - 1, 10) {}
                Dim num2 As Integer = 0
                Dim upperBound2 As Integer = arrSourceLines.GetUpperBound(0)
                For j As Integer = 0 To upperBound2
                    Dim left As String = arrSourceLines(j)
                    Dim flag3 As Boolean = Operators.CompareString(left, "", False) <> 0
                    If flag3 Then
                        array2(num2, 2) = Me.MyTRIM(Strings.Mid(arrSourceLines(j), 9, 8))
                        array2(num2, 3) = ErstelldatumTTMMJJ
                        array2(num2, 4) = Me.MyTRIM(Strings.Mid(arrSourceLines(j), 18, 13))
                        array2(num2, 6) = Me.Shorten(Strings.Mid(arrSourceLines(j), 73, 11), 7)
                        array2(num2, 8) = Me.Expand(Strings.Trim(Strings.Mid(arrSourceLines(j), 31, 42)), 10)
                        num2 += 1
                    End If
                Next
                array = Me.DeleteEntries1(array2)
            Catch expr_139 As Exception
                ProjectData.SetProjectError(expr_139)
                array = Nothing
                ProjectData.ClearProjectError()
            End Try
        End If
        Return array
    End Function

    Private Function ReadNewDATData(arrSourcelines As String(), ErstelldatumTTMMJJ As String) As String(,)
        ' The following expression was wrapped in a checked-statement
        Dim array3 As String(,)
        Try
            Dim num As Integer = -1
            Dim upperBound As Integer = arrSourcelines.GetUpperBound(0)
            For i As Integer = 0 To upperBound
                Dim flag As Boolean = Operators.CompareString(arrSourcelines(i), "", False) <> 0
                If flag Then
                    num += 1
                End If
            Next
            Dim array As String(,) = New String(num + 1 - 1, 10) {}
            Dim num2 As Integer = 0
            Dim upperBound2 As Integer = arrSourcelines.GetUpperBound(0)
            For j As Integer = 0 To upperBound2
                Dim array2 As String() = Strings.Split(arrSourcelines(j), ";", -1, CompareMethod.Binary)
                Dim flag2 As Boolean = Operators.CompareString(array2(0), "", False) <> 0
                If flag2 Then
                    Dim flag3 As Boolean = Operators.CompareString(array(num2, 2), "1678", False) = 0 And Operators.CompareString(array(num2, 6), "222", False) = 0
                    If flag3 Then
                        Debugger.Break()
                    End If
                    array(num2, 2) = array2(2)
                    array(num2, 3) = ErstelldatumTTMMJJ
                    array(num2, 4) = array2(4)
                    array(num2, 6) = Me.Expand(array2(6), 7)
                    array(num2, 8) = Me.Expand(array2(8), 10)
                    num2 += 1
                End If
            Next
            array3 = Me.DeleteEntries1(array)
            array3 = Me.ChangeArticleNumbers(array3)
        Catch expr_12D As Exception
            ProjectData.SetProjectError(expr_12D)
            array3 = Nothing
            ProjectData.ClearProjectError()
        End Try
        array3 = array3
        Return array3
    End Function

    Private Function DeleteEntries1(DATSource As String(,)) As String(,)
        ' The following expression was wrapped in a checked-statement
        Dim array As String(,) = New String(DATSource.GetUpperBound(0) + 1 - 1, DATSource.GetUpperBound(1) + 1 - 1) {}
        Dim num As Integer = 0
        LogMessage.LogOnly("Löschen von nicht benötigten Kunden- und Artikeldaten laut loeschKunde.txt & loeschArtikel.txt.")
        Dim upperBound As Integer = DATSource.GetUpperBound(0)
        For i As Integer = 0 To upperBound
            Dim flag As Boolean = False
            Try
                Dim enumerator As List(Of String).Enumerator = Module1.ListDelCustomer.GetEnumerator()
                While enumerator.MoveNext()
                    Dim current As String = enumerator.Current
                    Dim flag2 As Boolean = Operators.CompareString(Me.MyTRIM(DATSource(i, 2)), Me.MyTRIM(current), False) = 0
                    If flag2 Then
                        flag = True
                    End If
                End While
            Finally
                Dim enumerator As List(Of String).Enumerator
                CType(enumerator, IDisposable).Dispose()
            End Try
            Try
                Dim enumerator2 As List(Of String).Enumerator = Module1.ListDelArticle.GetEnumerator()
                While enumerator2.MoveNext()
                    Dim current2 As String = enumerator2.Current
                    Dim flag3 As Boolean = Operators.CompareString(Me.MyTRIM(DATSource(i, 8)), Me.MyTRIM(current2), False) = 0
                    If flag3 Then
                        flag = True
                    End If
                End While
            Finally
                Dim enumerator2 As List(Of String).Enumerator
                CType(enumerator2, IDisposable).Dispose()
            End Try
            Dim flag4 As Boolean = Not flag
            If flag4 Then
                Dim upperBound2 As Integer = DATSource.GetUpperBound(1)
                For j As Integer = 0 To upperBound2
                    array(num, j) = DATSource(i, j)
                Next
                num += 1
            End If
        Next
        Dim array2 As String(,) = New String(num - 1 + 1 - 1, array.GetUpperBound(1) + 1 - 1) {}
        Dim num2 As Integer = 0
        Dim upperBound3 As Integer = array.GetUpperBound(0)
        For k As Integer = 0 To upperBound3
            Dim flag5 As Boolean = Not Information.IsNothing(array(k, 2))
            If flag5 Then
                Dim upperBound4 As Integer = array.GetUpperBound(1)
                For l As Integer = 0 To upperBound4
                    array2(num2, l) = array(k, l)
                Next
                num2 += 1
            End If
        Next
        Return array2
    End Function

    Private Function ChangeArticleNumbers(DatSource As String(,)) As String(,)
        LogMessage.LogOnly("Austauschen von Artikelnummern laut tauscheArtikel.txt.")
        Dim flag As Boolean = Not Information.IsNothing(Module1.ListChangeArticle)
        ' The following expression was wrapped in a checked-statement
        Dim result As String(,)
        If flag Then
            Dim array As String(,) = New String(Module1.ListChangeArticle.Count - 1 + 1 - 1, 1) {}
            Try
                Dim upperBound As Integer = array.GetUpperBound(0)
                For i As Integer = 0 To upperBound
                    Dim array2 As String() = Strings.Split(Module1.ListChangeArticle(i), ";", -1, CompareMethod.Binary)
                    array(i, 0) = Me.MyTRIM(array2(0))
                    array(i, 1) = Me.MyTRIM(array2(1))
                Next
                Dim upperBound2 As Integer = DatSource.GetUpperBound(0)
                For j As Integer = 0 To upperBound2
                    Dim upperBound3 As Integer = array.GetUpperBound(0)
                    For k As Integer = 0 To upperBound3
                        Dim flag2 As Boolean = Operators.CompareString(DatSource(j, 8), Me.Expand(array(k, 0), 10), False) = 0
                        If flag2 Then
                            DatSource(j, 8) = Me.Expand(array(k, 1), 10)
                        End If
                    Next
                Next
            Catch expr_111 As Exception
                ProjectData.SetProjectError(expr_111)
                ProjectData.ClearProjectError()
            End Try
            result = DatSource
        Else
            result = DatSource
        End If
        Return result
    End Function

    Private Function WriteSedasData() As Boolean
        LogMessage.LogOnly("Schreiben der Sedas.dat...")
        Dim list As List(Of String) = New List(Of String)()
        Dim result As Boolean = False
        Me._SedasHeader = String.Concat(New String() {"010()000377777777777771", Me._ErstelldatumSedas, ";,", Conversions.ToString(Me._Counter), vbCrLf & ";)0240051310000002"})
        list.Add(Me._SedasHeader)
        Dim i As Integer = 0
        Dim num As Integer = 0
        Dim upperBound As Integer = Me._DATContent.GetUpperBound(0)
        ' The following expression was wrapped in a checked-statement
        Try
            While i <= Me._DATContent.GetUpperBound(0)
                list.Add(String.Concat(New String() {";030,14,00000000000000000,", Me._DATContent(i, 3), ",", Me.ReverseDate(Me._DATContent(i, 4)), ",,,,", Me._DATContent(i, 2), "         ,,"}))
                Me._Customers += 1
                Dim text As String = "0"
                While Operators.CompareString(Me._DATContent(i, 2), Me._DATContent(num, 2), False) = 0
                    Me._DataSets += 1
                    list.Add(String.Concat(New String() {";040000", Me._DATContent(num, 8), ",4", Me._DATContent(num, 6), ",,,,02 000000,,"}))
                    text = Conversions.ToString(Conversions.ToInteger(text) + Conversions.ToInteger(Me._DATContent(num, 6)))
                    num += 1
                    Dim flag As Boolean = num > Me._DATContent.GetUpperBound(0)
                    If flag Then
                        Exit While
                    End If
                End While
                Dim num2 As Integer = 12 - Strings.Len(text)
                For j As Integer = 1 To num2
                    text = "0" + text
                Next
                list.Add(";05" + text)
                i = num
                Me._SummeGes += Conversions.ToInteger(text)
            End While
            Me._SedasFooter = String.Concat(New String() {";06", Me.Expand(Conversions.ToString(Me._Customers), 3), ",", Me.Expand(Conversions.ToString(Me._DataSets), 4), vbCrLf & ";07000000,00001,00001,000000,("})
            list.Add(Me._SedasFooter)
            Dim flag2 As Boolean = Strings.InStr(Me._DestinationPath, "\", CompareMethod.Binary) > 0
            If flag2 Then
                Dim flag3 As Boolean = Not File.Exists(Me._DestinationPath)
                If flag3 Then
                    Dim flag4 As Boolean = Not Directory.Exists(Strings.Mid(Me._DestinationPath, 1, Strings.InStrRev(Me._DestinationPath, "\", -1, CompareMethod.Binary)))
                    If flag4 Then
                        Directory.CreateDirectory(Strings.Mid(Me._DestinationPath, 1, Strings.InStrRev(Me._DestinationPath, "\", -1, CompareMethod.Binary)))
                    End If
                End If
            Else
                Me._DestinationPath = Directory.GetCurrentDirectory() + "\" + Me._DestinationPath
            End If
            Using streamWriter As StreamWriter = New StreamWriter(Me._DestinationPath, False)
                Try
                    Dim enumerator As List(Of String).Enumerator = list.GetEnumerator()
                    While enumerator.MoveNext()
                        Dim current As String = enumerator.Current
                        streamWriter.WriteLine(current)
                    End While
                Finally
                    Dim enumerator As List(Of String).Enumerator
                    CType(enumerator, IDisposable).Dispose()
                End Try
                streamWriter.WriteLine("                                                                                    ")
            End Using
            result = True
        Catch expr_39B As Exception
            ProjectData.SetProjectError(expr_39B)
            Dim ex As Exception = expr_39B
            LogMessage.LogOnly(ex.ToString())
            Interaction.MsgBox(ex.ToString(), MsgBoxStyle.OkOnly, Nothing)
            result = False
            ProjectData.ClearProjectError()
        End Try
        Return result
    End Function

    Private Function ReverseDate(str As String) As String
        Dim text As String = ""
        Dim flag As Boolean = Operators.CompareString(str, "", False) <> 0
        ' The following expression was wrapped in a checked-statement
        If flag Then
            str = Strings.Trim(str)
            Dim num As Integer = Strings.Len(str) - 1
            For i As Integer = num To 1 Step -2
                Dim str2 As String = Strings.Mid(str, i, 2)
                text += str2
            Next
        End If
        Return text
    End Function

    Private Function MyTRIM(MyText As String) As String
        MyText = Strings.Trim(MyText)
        While Operators.CompareString(Strings.Mid(MyText, 1, 1), "0", False) = 0
            MyText = Strings.Mid(MyText, 2)
        End While
        Return MyText
    End Function

    Private Function Shorten(input As String, limit As Integer) As String
        Dim num As Integer = Strings.Len(input)
        Dim flag As Boolean = num > limit
        Dim result As String
        If flag Then
            ' The following expression was wrapped in a checked-expression
            result = Strings.Mid(input, num - limit + 1)
        Else
            result = input
        End If
        Return result
    End Function

    Private Function Expand(input As String, limit As Integer) As String
        Dim num As Integer = Strings.Len(input)
        Dim text As String = ""
        Dim flag As Boolean = num < limit
        ' The following expression was wrapped in a checked-statement
        If flag Then
            Dim num2 As Integer = limit - num
            For i As Integer = 1 To num2
                text += "0"
            Next
            text += input
        Else
            text = input
        End If
        Return text
    End Function
End Class

File: DTAtoSEDAS/DatToSedas_CSharp/ConvertDatToSedas.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DatToSedas_CSharp
{
    class ConvertDatToSedas
    {
        private string _SourcePath;
        private string _SourceData;
        private string _DestinationPath;
        private string _DestinationData;
        private string _SedasHeader;
        private string _SedasFooter;
        private string _BlockHeader;
        private string[,] _DATContent;
        private int _DataSets;
        private int _Customers;
        private int _SummeGes;
        private List<string> _ListDelCustomer = new List<string>();
        private List<string> _ListDelArticle = new List<string>();
        private int _Counter;
        private string _ErstelldatumSedas;


        //KONSTRUKTOR
        public ConvertDatToSedas(string src, string dst, int cnt, List<string> dC = null, List<string> dA = null)
        {
            this._SourcePath = "";
            this._SourceData = "";
            this._DestinationPath = "";
            this._DestinationData = "";
            this._SedasHeader = "";
            this._SedasFooter = "";
            this._BlockHeader = "";
            this._DataSets = 0;
            this._Customers = 0;
            this._SummeGes = 0;
            this._Counter = 0;
            this._ErstelldatumSedas = "";
            this._SourcePath = src;
            this._DestinationPath = dst;
            this._Counter = cnt;
            this._ListDelArticle = dA;
            this._ListDelCustomer = dC;
        }

        public bool ConvertFile()
        {
            LogMessage.LogOnly("Beginn der Konvertierung...");
            this._ErstelldatumSedas = this.ReverseDate(Strings.Mid(DateAndTime.Now.ToString(), 1, 10).Replace(".", "").Remove(4, 2));

            try
            {
                if (!File.Exists(_SourcePath))
                {
                    return false; ;
                }
                LogMessage.LogOnly("Einlesen der Quelldatei...");

                using (StreamReader sr = new StreamReader(_SourcePath))
                {
                    this._SourceData = sr.ReadToEnd();
                }
            }
            catch (Exception ex)
            {
                return false;
            }

            string[] array = Strings.Split(this._SourceData, vbCrLf, -1, CompareMethod.Binary);
            int num = 0;
            int upperBound = array.GetUpperBound(0);
            // The following expression was wrapped in a checked-statement
            for (int i = 0; i <= upperBound; i++)
            {
                if (array[i] == "")
                {
                    num += 1;
                }
            }

            string[] array2 = new string[num] { }; //TODO  array mit festen dimensionen initialisieren
            int num2 = 0;
            int upperBound2 = array.GetUpperBound(0);
            for (int j = 0; j <= upperBound; j++)
            {
                if (array[j] == "")
                {
                    array2[num2] = array[j];
                    num2 += 1;
                }
            }
            array = array2;

            try
            {
                bool flag4 = Operators.CompareString(Strings.Mid(array(0), 1, 2), "NF", false) = 0;
                if (flag4)
                {
                    LogMessage.LogOnly("Einlesen neues Dateiformat...");
                    this._DATContent = this.ReadNewDATData(array, this._ErstelldatumSedas);
                }
                else
                {
                    LogMessage.LogOnly("Einlesen altes Dateiformat...");
                    this._DATContent = this.ReadDATData(array, this._ErstelldatumSedas);
                }

                bool flag5 = Not Information.IsNothing(this._DATContent) And this.WriteSedasData();
                if (flag5)
                {
                    LogMessage.LogOnly("Konvertierung in SEDAS.DAT abgeschlossen.");
                    return true;
                }
            }
            catch (Exception ex)
            {
                LogMessage.LogOnly("Fehler beim Konvertieren in Sedas.dat." & vbCrLf + ex.ToString());

                return false;
            }
            return false;
        }


        private string[,] ReadDATData(string[] arrSourceLines, string ErstelldatumTTMMJJ)
        {
            bool flag = Operators.CompareString(arrSourceLines(0), "", false) = 0

                // The following expression was wrapped in a checked-statement
            string[,] array;
            if (arrSourceLines[0] == "")
            {
                array = null;
                array = array;
            }
            else
            {
                try
                {
                    int num = -1;
                    int upperBound = arrSourceLines.GetUpperBound(0);
                    for (int i = 0; i <= upperBound; i++)
                    {
                        bool flag2 = Operators.CompareString(arrSourceLines(i), "", false) <> 0;
                        if (flag2)
                            num += 1;
                    }

                    string[,] array2 = New String(num + 1 - 1, 10) { }; //Array dimensionieren
                    int num2 = 0;
                    int upperBound2 = arrSourceLines.GetUpperBound(0);
                    for (int j = 0; j < upperBound2; j++)
                    {
                        string left = arrSourceLines(j);
                        bool flag3 = Operators.CompareString(left, "", false) <> 0;
                        if (flag3)
                        {
                            array2(num2, 2) = this.MyTRIM(Strings.Mid(arrSourceLines(j), 9, 8));
                            array2(num2, 3) = ErstelldatumTTMMJJ;
                            array2(num2, 4) = this.MyTRIM(Strings.Mid(arrSourceLines(j), 18, 13));
                            array2(num2, 6) = this.Shorten(Strings.Mid(arrSourceLines(j), 73, 11), 7);
                            array2(num2, 8) = this.Expand(Strings.Trim(Strings.Mid(arrSourceLines(j), 31, 42)), 10);
                            num2 += 1;
                        }
                    }
                    array = this.DeleteEntries1(array2);
                }
                catch (Exception ex)
                {
                    array = null;
                }
            }
            return array;
        }


        private string[,] ReadNewDATData(string arrSourcelines, string ErstelldatumTTMMJJ)
        {
            // The following expression was wrapped in a checked-statement
            string[,] array3 = null;
            try
            {
                int num = -1;
                int upperBound = arrSourcelines.GetUpperBound(0);
                for (int i = 0; i <= upperBound; i++)
                {
                    bool flag = Operators.CompareString(arrSourcelines(i), "", false) <> 0;
                    if (flag)
                    {
                        num += 1;
                    }
                }

                string[,] array = new string[,] { };// (num + 1 - 1, 10) { };
                int num2 = 0;
                int upperBound2 = arrSourcelines.GetUpperBound(0);
                for (int j = 0; j <= upperBound2; j++)
                {
                    string[] array2 = Strings.Split(arrSourcelines(j), ";", -1, CompareMethod.Binary);
                    bool flag2 = Operators.CompareString(array2(0), "", false) <> 0;
                    if (flag2)
                    {
                        bool flag3 = Operators.CompareString(array(num2, 2), "1678", false) = 0 And Operators.CompareString(array(num2, 6), "222", false) = 0;
                        if (flag3)
                        {
                            Debugger.Break();
                        }
                        array(num2, 2) = array2(2);
                        array(num2, 3) = ErstelldatumTTMMJJ;
                        array(num2, 4) = array2(4);
                        array(num2, 6) = this.Expand(array2(6), 7);
                        array(num2, 8) = this.Expand(array2(8), 10);
                        num2 += 1;
                    }
                }
                array3 = this.DeleteEntries1(array);
                array3 = this.ChangeArticleNumbers(array3);
            }
            catch (Exception ex)
            {
            }
            return array3;
        }

        private string[,] DeleteEntries1(string[,] DATSource)
        {
            // The following expression was wrapped in a checked-statement
            string[,] array = New String(DATSource.GetUpperBound(0) + 1 - 1, DATSource.GetUpperBound(1) + 1 - 1) { };
            int num = 0;
            LogMessage.LogOnly("Löschen von nicht benötigten Kunden- und Artikeldaten laut loeschKunde.txt & loeschArtikel.txt.");
            int upperBound = DATSource.GetUpperBound(0);
            for ()// i As Integer = 0 To upperBound
            {
                bool flag = false;
                try
                {
                    Dim enumerator As List(Of String).Enumerator = Module1.ListDelCustomer.GetEnumerator();
                    while ()// enumerator.MoveNext()
                    {
                        string  current = enumerator.Current;
                        bool flag2 = Operators.CompareString(this.MyTRIM(DATSource(i, 2)), this.MyTRIM(current), false) = 0;
                        if (flag2)
                        {
                            flag = true;
                        }
                    }
                }
                finally
                {
                    Dim enumerator As List(Of String).Enumerator;
                    CType(enumerator, IDisposable).Dispose();
                }

                try
                {
                    Dim enumerator2 As List(Of String).Enumerator = Module1.ListDelArticle.GetEnumerator();
                    while ()// enumerator2.MoveNext()
                    {
                        string current2 enumerator2.Current;
                        bool flag3  = Operators.CompareString(this.MyTRIM(DATSource(i, 8)), this.MyTRIM(current2), false) = 0;
                        if (flag3)
                        {
                            flag = true;
                        }
                    }
                }
                finally
                {
                    Dim enumerator2 As List(Of String).Enumerator;
                    CType(enumerator2, IDisposable).Dispose();
                }

                bool flag4  = Not flag;
                if (flag4)
                {
                    int upperBound2 DATSource.GetUpperBound(1);
                    for ()// j As Integer = 0 To upperBound2
                    {
                        array(num, j) = DATSource(i, j);
                    }
                    num += 1;
                }
            }

            Dim array2 As String(,) = New String(num - 1 + 1 - 1, array.GetUpperBound(1) + 1 - 1) { };
            int num2 = 0;
            int upperBound3 = array.GetUpperBound(0);
            for ()// k As Integer = 0 To upperBound3
            {
                bool flag5 = Not Information.IsNothing(array(k, 2));
                if (flag5)
                {
                    int upperBound4 = array.GetUpperBound(1);
                    for ()// l As Integer = 0 To upperBound4
                    {
                        array2(num2, l) = array(k, l);
                    }
                    num2 += 1;
                }
            }
            return array2;
        }

        private string[,] ChangeArticleNumbers(DatSource As String(,))
        {
            LogMessage.LogOnly("Austauschen von Artikelnummern laut tauscheArtikel.txt.");
            bool flag = Not Information.IsNothing(Module1.ListChangeArticle);

            //The following expression was wrapped in a checked-statement
            string[,] result;
            if (flag)
            {
                string[,] array = New String(Module1.ListChangeArticle.Count - 1 + 1 - 1, 1) { };
                try
                {
                    int upperBound = array.GetUpperBound(0);
                    for (int i = 0; i <= upperBound; i++) // i As Integer = 0 To upperBound
                    {
                        Dim array2 As String() = Strings.Split(Module1.ListChangeArticle(i), ";", -1, CompareMethod.Binary);
                        array(i, 0) = this.MyTRIM(array2(0));
                        array(i, 1) = this.MyTRIM(array2(1));
                    }
                    int upperBound2 = DatSource.GetUpperBound(0);
                    for (int j = 0; j <= upperBound2; j++) // j As Integer = 0 To upperBound2
                    {
                        int upperBound3 = array.GetUpperBound(0);
                        for (int k = 0; k <= upperBound3; k++) // k As Integer = 0 To upperBound3
                        {
                            bool flag2 = Operators.CompareString(DatSource(j, 8), this.Expand(array(k, 0), 10), false) = 0;
                            if (flag2)
                            {
                                DatSource(j, 8) = this.Expand(array(k, 1), 10);
                            }
                        }
                    }
                }
                catch (Exception ex)
                {
                }
            }
            else
            {
                result = DatSource;
            }

            result = DatSource;
        }





            return result;
        }

    private bool WriteSedasData()
    {
        LogMessage.LogOnly("Schreiben der Sedas.dat...");
        List<string> list = new List<string>();
        bool result = false;
        this._SedasHeader = String.Concat(New String() { "010()000377777777777771", this._ErstelldatumSedas, ";,", Conversions.ToString(this._Counter), vbCrLf & ";)0240051310000002"})  ;
        list.Add(this._SedasHeader);
        int i = 0;
        int j = 0;
        int upperBound = this._DATContent.GetUpperBound(0);


        // The following expression was wrapped in a checked-statement
        try
        {
            while (i <= this._DATContent.GetUpperBound(0))
            {
                list.Add(String.Concat(New String() { ";030,14,00000000000000000,", this._DATContent(i, 3), ",", this.ReverseDate(this._DATContent(i, 4)), ",,,,", this._DATContent(i, 2), "         ,,"}));
                this._Customers += 1;
                Dim text As String = "0";


                while (Operators.CompareString(this._DATContent(i, 2), this._DATContent(num, 2), false) = 0)
                {
                    this._DataSets += 1;
                    list.Add(String.Concat(New String() { ";040000", this._DATContent(num, 8), ",4", this._DATContent(num, 6), ",,,,02 000000,,"}));
                    text = Conversions.ToString(Conversions.ToInteger(text) + Conversions.ToInteger(this._DATContent(num, 6)));
                    num += 1;
                    Dim flag As Boolean = num > this._DATContent.GetUpperBound(0);

                    if (flag)
                    {
                        Exit While;
                    }
                }

                int num2 = 12 - Strings.Len(text);
                for (int j = 0; j <= num; j++) // j As Integer = 1 To num2
                {
                    text = "0" + text;
                }
                list.Add(";05" + text);
                i = num;
                this._SummeGes += Conversions.ToInteger(text);
            }

            this._SedasFooter = String.Concat(New String() { ";06", this.Expand(Conversions.ToString(this._Customers), 3), ",", this.Expand(Conversions.ToString(this._DataSets), 4), vbCrLf & ";07000000,00001,00001,000000,("}) ;
            list.Add(this._SedasFooter);
            Dim flag2 As Boolean = Strings.InStr(this._DestinationPath, "\\", CompareMethod.Binary) > 0;


            if (flag2)
            {
                Dim flag3 As Boolean = Not File.Exists(this._DestinationPath);
                if (flag3)
                {
                    bool flag4 = Not Directory.Exists(Strings.Mid(this._DestinationPath, 1, Strings.InStrRev(this._DestinationPath, "\\", -1, CompareMethod.Binary)));


                    if (flag4)
                    {
                        Directory.CreateDirectory(Strings.Mid(this._DestinationPath, 1, Strings.InStrRev(this._DestinationPath, "\\", -1, CompareMethod.Binary)));
                    }
                }
            }
            else
            {
                this._DestinationPath = Directory.GetCurrentDirectory() + "\\" + this._DestinationPath;
            }


            using (StreamWriter sw = new StreamWriter(this._DestinationPath, false))
            {
                try
                {
                    Dim enumerator As List(Of String).Enumerator = list.GetEnumerator();
                    while (enumerator.MoveNext())
                    {
                        Dim current As String = enumerator.Current;
                        streamWriter.WriteLine(current);
                    }
                }
                finally
                {
                    Dim enumerator As List(Of String).Enumerator;
                    CType(enumerator, IDisposable).Dispose();
                }

                sw.WriteLine("                                                                                    ");
            }
            result = true;


        }
        catch (Exception ex)
        {
            LogMessage.LogOnly(ex.ToString());
            result = false;
        }

        return result;
    }

    private string ReverseDate(string str)
    {
        string text = "";
        bool flag = Operators.CompareString(str, "", false) <> 0;

        //The following expression was wrapped in a checked-statement
        if (flag)
            str = Strings.Trim(str);
        int num = Strings.Len(str) - 1;
        for (int i = 0; i >= num; i = i - 2) //i As Integer = num To 1 Step - 2
        {
            Dim str2 As String = Strings.Mid(str, i, 2);
            text += str2;
        }
        return text;
    }

    private string MyTRIM(string MyText)
    {
        MyText = Strings.Trim(MyText);
        while (Operators.CompareString(Strings.Mid(MyText, 1, 1), "0", false) = 0)
        {
            MyText = Strings.Mid(MyText, 2);
        }
        return MyText;
    }

    private string Shorten(string input, int limit)
    {
        Dim num As Integer = Strings.Len(input)
                            Dim flag As Boolean = num > limit
                            Dim result As String
                            If flag Then
                                ' The following expression was wrapped in a checked-expression
                                result = Strings.Mid(input, num - limit + 1)
                            Else
            result = input
                            End If







            return result;
    }

    private string Expand(string input, int limit)
    {
        int num = input.Length;
        string text = "";
        bool flag = num < limit;

        // The following expression was wrapped in a checked-statement                
        if (flag)
        {
            int num2 = limit - num;
            for (int i = 0; i <= num2; i++)// i As Integer = 1 To num2
            {
                text += "0";
            }
            text += input;
        }
        else
        {
            text = input;
        }
        return text;
    }
}
}